src/parseTools.js

```diff
@@ -1043,3 +1043,12 @@ function getPerformanceNow() {
 function implicitSelf() {
   return ENVIRONMENT.includes('node') ? 'self.' : '';
 }
+
+function ENVIRONMENT_IS_MAIN_THREAD() {
+  var envs = [];
+  if (USE_PTHREADS) envs.push('ENVIRONMENT_IS_PTHREAD');
```

Comment:
Note: this -> PTHREADS with latest emscripten main, otherwise the PR works well!